src/NosnocNLP.m

```diff
@@ -236,6 +255,18 @@ function mpcc_to_nlp(obj, sigma_p)
                 mpcc.ubw(mpcc.ind_s_terminal),...
                 mpcc.w0(mpcc.ind_s_terminal));
 
+            % Add s_numerical
+           obj.addPrimalVector(mpcc.w(mpcc.ind_s_numerical),...
+                mpcc.lbw(mpcc.ind_s_numerical),...
```

Comment:
Why is this one done again. Compared to ~30 lines earlier?

Reply:
had to dig a bit here. This is because this is done both at the mpcc and control stage level. this is one of those things that needs to be done this way in order to maintain order of variables (which is something that needs to be part of a redesign.)